tools/scripts/refresh-ubuntu-amis.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+
+"""
+Fetch the latest official Ubuntu AMIs that we hardcode into our system.
+
+This will update all *_master_ami and *_bastion_ami tags in bumpenvs.yaml file.
+
+Usage: refresh-ubuntu-amis.py path/to/bumpenvs.yaml
+"""
+
+import sys
+from typing import Dict, List
+
+import requests
+import yaml
+
+get_cache = {}  # type: Dict[str, bytes]
+
+
+def cacheable_get(url: str) -> bytes:
+    global get_cache
+    if url not in get_cache:
+        req = requests.get(url)
+        assert req.ok, f"request to {url} failed with HTTP {req.status_code}"
+        get_cache[url] = req.content
+    return get_cache[url]
+
+
+def get_ubuntu_ami(release: str, region: str) -> str:
+    resp = cacheable_get(
+        f"https://cloud-images.ubuntu.com/query/{release}/server/released.current.txt"
+    ).decode("utf8")
+
+    ami_lines = [line.split("\t") for line in resp.rstrip("\n").split("\n")]
```

Comment:
non-blocking: `resp.splitlines()` instead (unless you specifically want to account for the possibility of multiple trailing newlines).

non-blocking: Use of `namedtuple` would be pleasing (but not very necessary due to the small scope here).

Reply:
re: splitlines: there are too many builtin methods in python.  But I'm glad to learn about this one.

re namedtuple: There's too many fields that I don't know how to name... I thought this came out nicer (I tried namedtuple first actually)